pkg/upgrade/v1_15_0_test.go

```diff
@@ -0,0 +1,65 @@
+package upgrade
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	v1 "github.com/jaegertracing/jaeger-operator/pkg/apis/jaegertracing/v1"
+)
+
+func TestUpgradeDeprecatedOptions(t *testing.T) {
+	nsn := types.NamespacedName{Name: "my-instance"}
+	existing := v1.NewJaeger(nsn)
+	existing.Status.Version = "1.14.0"
+	existing.Spec.Collector.Options = v1.NewOptions(map[string]interface{}{
+		"collector.host-port": "jaeger.example.com:14268",
+	})
+	objs := []runtime.Object{existing}
+
+	s := scheme.Scheme
+	s.AddKnownTypes(v1.SchemeGroupVersion, &v1.Jaeger{})
+	s.AddKnownTypes(v1.SchemeGroupVersion, &v1.JaegerList{})
+	cl := fake.NewFakeClient(objs...)
+
+	// test
+	assert.NoError(t, ManagedInstances(cl))
+
+	// verify
+	persisted := &v1.Jaeger{}
+	assert.NoError(t, cl.Get(context.Background(), nsn, persisted))
+	assert.Equal(t, latest.v, persisted.Status.Version)
+
+	opts := persisted.Spec.Collector.Options.Map()
+	assert.Contains(t, opts, "reporter.tchannel.host-port")
+	assert.Equal(t, "jaeger.example.com:14268", opts["reporter.tchannel.host-port"])
+	assert.NotContains(t, opts, "collector.host-port")
+}
+
+func TestRemoveDeprecatedFlagWithNoReplacement(t *testing.T) {
+	// prepare
+	nsn := types.NamespacedName{Name: "my-instance"}
+	existing := v1.NewJaeger(nsn)
+	existing.Status.Version = "1.14.0"
+	existing.Spec.Collector.Options = v1.NewOptions(map[string]interface{}{
+		"cassandra.enable-dependencies-v2": "true",
+	})
+
+	// sanity check
+	assert.Contains(t, existing.Spec.Collector.Options.Map(), "cassandra.enable-dependencies-v2")
+	assert.Len(t, existing.Spec.Collector.Options.Map(), 1)
+
+	// test
+	updated, err := upgrade1_15_0(nil, *existing)
```

Comment:
Wondering why this approach wasn't also used in the previous test? Not necessarily an issue - but this focuses on the option transformation rather than the upgrade mechanism, which I think is appropriate.

Reply:
One test is also covering all the wiring from higher level parts and that the change is being persisted. We don't need to run this whole chain all the time, so, the second test is a more like a *unit* test, testing the function alone.

Reply:
Ok sound fine then.